src/formatDate.js

```diff
@@ -8,7 +8,30 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const dateMap = {};
+
+  fromFormat.slice(0, 3).forEach((part, index) => {
+    dateMap[part] = dateParts[index];
```

Comment:
Ensure that the date parts are correctly mapped even if the `fromFormat` array is not in the order of 'YYYY', 'MM', 'DD'. The current implementation assumes a specific order, which may not always be the case.

Reply:
done